### PR TITLE
Fixed the application of SortBy string to IQueryable

### PR DESCRIPTION
**Pull request by rurku**
You almost had it right @HerbertBodner. You just need to reverse the sort fields.

### CODE REVIEW

**Comment by HerbertBodner** [comment]
Well, this was not how we planned to do it. The idea was to use expressions instead of strings in the GetDbSortField. Then you get a build error in case you rename a property in the entity class and you have to fix that. At the moment such changes build just fine, but it won't work at runtime.

**Comment by rurku** [comment]
Now I made it so that you configure the mapping in IoC. I don't really like it but I didn't know where to put it.

File: src/WaCore.Crud.Utils/LinqExtensions.cs [hunk 1]
```diff
@@ -3,22 +3,49 @@
...
+                if (sortColumnDescriptors == null)
+                {
+                    throw new ArgumentException($"Invalid sort field \"{orderItem.FieldName}\".", sortString);
+                }
+                else
+                {
+                    foreach (var sortDescriptor in sortColumnDescriptors.Reverse())
+                    {
+                        var asc = sortDescriptor.Asc ^ orderItem.OrderDirection == OrderItem.OrderBy.Descending;
+                        var methodName = asc ? "OrderBy" : "OrderByDescending";
```
Comment (yshchohaleu) [inline src/WaCore.Crud.Utils/LinqExtensions.cs]: Is it possible to use ThenBy here if there are several sort fields? Or we do not need it?

File: src/WaCore.Crud.Utils/Sorting/SortFieldMapping.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            Asc = asc;
+            return this;
+        }
+
+        public Type KeyType { get; set; }
+        // This should be of type Expression<Func<TEntity,KeyType>>
+        public Expression KeySelector { get; set; }
+        public bool Asc { get; set; }
+
+        public IQueryable<TEntity> Apply(IQueryable<TEntity> q, bool reverseOrder)
```
Comment (yshchohaleu) [inline src/WaCore.Crud.Utils/Sorting/SortFieldMapping.cs]: Do we need this method, it's not used as far as I understand?

**Comment by HerbertBodner** [comment]
I changed the target branch to master, because the previous target branch BaseControllers02 has already been merged to master.

File: src/WaCore.Crud.Utils/LinqExtensions.cs (hunk 1 above)
Comment (rurku) [reply src/WaCore.Crud.Utils/LinqExtensions.cs]: As discussed, just OrderBy and OrderByDescending work correctly when used in reverse order.

File: src/WaCore.Crud.Utils/Sorting/SortFieldMapping.cs (hunk 2 above)
Comment (rurku) [reply src/WaCore.Crud.Utils/Sorting/SortFieldMapping.cs]: Removed

**Pull request merged**